### DISCUSSION TITLE
Unable to get magick mogrify -trim and -fuzz to work

### DISCUSSION THREAD

**Discussion by dmcnaugh15** [ImageMagick/ImageMagick#7851] [Help] [answered]
### ImageMagick version

7.1.1-41-Q16-HDRI-x64-dll.exe

### Operating system, version and so on

Windows 11

### Description

Disclaimer: Very new to ImageMagick.  On the steep part of the learning curve.

Issue: I have a scanned rectangular photo with a white edge.  All I want to do is remove the edge.  Attached is the photo I'm working with.  Below are the lines I've tried, and which have not worked:

```
	magick mogrify -trim *
	magick mogrify -fuzz *
	magick mogrify -trim -fuzz *
	magick mogrify -trim -fuzz
	magick mogrify -fuzz distance 5%
	magick mogrify -define trim:edges=north,south -fuzz 5% -trim
```

Can anyone point me in the right direction?

![Image](https://github.com/user-attachments/assets/f6c00bc9-e029-4368-b212-16be03331a72)

**Comment by fmw42** [comment]
For one image, you should just use magick, not magick mogrify, which is used to process a folder of images. Also your commands need syntax changes to supply an input and output. Put the fuzz XX% before the -trim.  Use about 15% minimum for the fuzz with JPG files due to its compression artifacts. So try

`magick image.jpg -fuzz 15% -trim +repage x.jpg`

on Windows double the % to %%

![x](https://github.com/user-attachments/assets/3d8642f1-9de0-46f8-b29b-0bac6ea0752a)

**Comment by dmcnaugh15** [comment]
@fmw42 Thanks. This is the effect I'm going for.  In my attempt to simplify the question for this topic, I omitted important context.  I am doing this for multiple photos in a directory.  So, I would prefer to use mogrify.  I did attempt to integrate your suggestions into a mogrify script, but am still not having any success.  This is what I've tried (not working), based on your feedback:

```
	magick mogrify -fuzz 15% -trim +repage
	magick mogrify -fuzz 15% -trim
	magick mogrify -fuzz 15%% -trim
	magick mogrify -fuzz 15%% -trim +repage
```

**Comment by fmw42** [comment]
Mogrify uses the current directory to get the images. You can specify any other directory for the output. You can also specify the output format.  Mogrify keeps the input filenames for the output. For Windows syntax try

```
cd to directory with images
magick mogrify -path any_existing_output_directory -format output_suffix -fuzz 15%% -trim +repage * (or *.suffx)
```

* will try to process every file of every suffix
*.suffix will only process files with the specified suffix

See https://usage.imagemagick.org/basics/#mogrify

If you want to process any one image, then 

```
cd to directory with image
magick mogrify -path any_existing_output_directory -format output_suffix -fuzz 15%% -trim +repage filename.suffix
```

**Comment by dmcnaugh15** [comment]
Thank you so much @fmw42.  I'm almost there.  It's working, but the output file is going to the same directory as the original directory (see screenshot 1), instead of the existing output directory, and there are some cmd line errors (see screenshot 2).

My attempted scripts look like this:
```
magick mogrify C:\Users\dmcna\Desktop\Editing\Trimmed\ -format jpg -fuzz 15%% -trim +repage *
magick mogrify C:\Users\dmcna\Desktop\Editing\Trimmed -format jpg -fuzz 15%% -trim +repage *
```

SCREENSHOT 1 (WINDOWS EXPLORER FOLDER - AFTER OPERATION):
![Screenshot 1](https://github.com/user-attachments/assets/4c8859fc-d7af-4313-a9dd-fab4e0c432c9)

SCREENSHOT 2 (CMD LINE ERROR):
![Screenshot 2](https://github.com/user-attachments/assets/ea2fde36-6251-43f8-9bdc-b4a827c5d715)

**Comment by fmw42** [comment, marked as answer]
You forgot the -path before your directory.

magick mogrify **-path** C:\Users\dmcna\Desktop\Editing\Trimmed\ -format jpg -fuzz 15%% -trim +repage *

**Comment by dmcnaugh15** [comment]
THANK YOU!!!
